p4runtime_sh/shell.py

```diff
@@ -2410,15 +2411,32 @@ def sniff(self, function=None, timeout=None):
         the function to every packet-in message.
         """
         msgs = []
-        while True:
-            try:
-                msgs.append(self.packet_in_queue.get(block=True, timeout=timeout))
-            except queue.Empty:
-                # No item available when timeout.
-                break
-            except KeyboardInterrupt:
-                # User sends an interrupt(e.g., Ctrl+C).
-                break
+
+        if timeout is not None and timeout < 0:
+            raise ValueError("Timeout can\'t be a negative number.")
+
+        if timeout is None:
+            while True:
+                try:
+                    msgs.append(self.packet_in_queue.get(block=True))
```

Comment:
@Yi-Tseng I'm not sure why we don't apply the `function` (if one was provided) directly here as we receive the packet

But that can be addressed in a subsequent PR...

Reply:
Right, we can actually invoke the function when we receive a packet-in message. This can be improved in the next PR